### PR TITLE
[163] - User can't create a seller account if he has a buyer account using the same e-mail. Allow it

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-163 User can't create a seller account if he has a buyer account using the same e-mail. Allow it](https://jjbcapital.atlassian.net/browse/NL-163)

### What

Now if user with existing user account can have two accounts (roles: buyer and seller). Creation and Deletion of accounts also works. 

### How

User signs up to the application. After the verification of email he can choose the role he wants to proceed with - Buyer or Seller. If he selects Seller for example - the Seller functionality is displayed for him.

User signs in to the application. After entering login and password user selects under which role he wants to sign in. If user already has Seller role for example and selects Seller - application opens as it worked before. If he selects Buyer - system should notify him that he does not have a Buyer role and it will ask user if he wants to sign in as a Buyer. If the user confirms - application opens with Buyer functionality.

If user has Buyer and Seller roles at the same time - he cannot participate in his own auctions.

image

image

### Affected Area

Basically every area was affected from BE and FE.

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] All Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by otash8110 — DISMISSED** [review]

**Comment by khasanrashidov** [comment]
@TsvirkoNikita you can review, @otash8110 Otabek reviewed this pr and approved (but the merge-base changed after approval, yeah, which can be easily solved by switching to/back to comparing branch)))

image

**Review by otash8110 — DISMISSED** [review]

File: src/Auctionify.Infrastructure/Identity/IdentityService.cs [hunk 1]
```diff
@@ -105,26 +113,219 @@ public async Task<LoginResponse> LoginUserAsync(LoginViewModel userModel)
...
+					Errors = new[] { "User is not found or deleted" },
+					IsSuccess = false,
+				};
+			}
+
+			var userRoles = await _userRoleDbContextService.GetUnpaginatedListAsync(
+				ur => ur.UserId == user.Id && !ur.IsDeleted
+			);
+
+			if (userRoles.Any())
```
Comment (TsvirkoNikita) [inline src/Auctionify.Infrastructure/Identity/IdentityService.cs:171]: Inside foreach you get same role from db multiple times. You can simplify this section:

```
var userRole = await _roleManager.FindByNameAsync(role);
if (userRoles.Any(ur => ur.RoleId == userRole!.Id))
{
    var token = await GenerateJWTTokenWithUserClaimsAsync(user, role);

    token.Role = role;
    token.UserId = user.Id;

    return new LoginResponse { IsSuccess = true, Result = token };
}
```

File: src/Auctionify.Infrastructure/Identity/IdentityService.cs [hunk 2]
```diff
@@ -105,26 +113,219 @@ public async Task<LoginResponse> LoginUserAsync(LoginViewModel userModel)
...
+			var userRoles = await _userRoleDbContextService.GetUnpaginatedListAsync(
+				ur => ur.UserId == user.Id
+			);
+
+			if (userRoles == null)
+			{
+				return new LoginResponse { IsSuccess = false, Message = "User has no roles" };
+			}
+
+			if (userRoles.Any())
```
Comment (TsvirkoNikita) [inline src/Auctionify.Infrastructure/Identity/IdentityService.cs:229]: Similar section as above, please, refactor it

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

File: src/Auctionify.Infrastructure/Identity/IdentityService.cs (hunk 1 above)
Comment (khasanrashidov) [reply src/Auctionify.Infrastructure/Identity/IdentityService.cs:171]: Yes, thank you for noticing) I will refactor it)

File: src/Auctionify.Infrastructure/Identity/IdentityService.cs (hunk 2 above)
Comment (khasanrashidov) [reply src/Auctionify.Infrastructure/Identity/IdentityService.cs:231]: refactored it

**Comment by khasanrashidov** [comment]
fixed specified issues in pr comments

**Review by TsvirkoNikita — DISMISSED** [review]

**Review by SaidganiUsm — APPROVED** [review]

**Review by TsvirkoNikita — APPROVED** [review]
